persist/sqldb/sqldb.go

```diff
@@ -101,6 +101,9 @@ func CreatePostGresDBSession(kubectlConfig kubernetes.Interface, namespace strin
 		return nil, "", err
 	}
 
+	// default for connMaxLifetime to 300 seconds
+	session.SetConnMaxLifetime(300 * time.Second)
```

Comment:
what is the default for this if unset?

Reply:
@alexec I don't think there's a default being set, tried looking for it in the codebase

Reply:
1. Does this change get undone by line 110 if if the `connMaxLifetime` setting is **_absent_** from their connectionPool stanza? e.g.

```yaml
persistence: |-
  connectionPool:
    maxIdleConns: 100
    maxOpenConns: 0
```

2. Related to that, is there a way for this to be unset? In other words, what would be the way to configure this *not* to have a max lifetime?

Reply:
The default would just be the pointer itself, so 0
0 means no limit on connection 
https://github.com/argoproj/argo-workflows/blob/master/docs/workflow-controller-configmap.yaml#L270

I'm thinking this should be good user practice to configure SQL connection instead of defaulting it from argo.
perhaps we could change and mention this in the configmap document

Reply:
here the default from db.v3
```
var DefaultSettings [Settings](https://pkg.go.dev/upper.io/db.v3#Settings) = &settings{
	preparedStatementCacheEnabled: 0,
	connMaxLife[time](https://pkg.go.dev/time):               time.[Duration](https://pkg.go.dev/time#Duration)(0),
	maxIdleConns:                  10,
	maxOpenConns:                  0,
}
```
// Set the maximum lifetime of a connection to 1 hour. Setting it to 0
// means that there is no maximum lifetime and the connection is reused
// forever (which is the default behavior).
db.SetConnMaxLifetime(time.Hour)